src/odb.c

```diff
@@ -84,19 +84,34 @@ static int odb_read_hardcoded(bool *found, git_rawobj *raw, const git_oid *id)
 	return 0;
 }
 
-int git_odb__format_object_header(char *hdr, size_t n, git_off_t obj_len, git_otype obj_type)
+int git_odb__format_object_header(
+	size_t *written,
+	char *hdr,
+	size_t hdr_size,
+	git_off_t obj_len,
+	git_otype obj_type)
 {
 	const char *type_str = git_object_type2string(obj_type);
-	int len = p_snprintf(hdr, n, "%s %lld", type_str, (long long)obj_len);
-	assert(len > 0 && len <= (int)n);
-	return len+1;
+	int hdr_max = (hdr_size > INT_MAX-2) ? (INT_MAX-2) : (int)hdr_size;
```

Comment:
Why don't you use the `MAX` macro here?

Reply:
Especially since this line is not a maximum, but a minimum of `hdr_size` and `INT_MAX-2`. So even though its not wrong, it's a bit surprising to the reader

Reply:
Because the cast it important in the assignment, and only in the assignment, not in the comparison.

Reply:
:+1:

Reply:
(Or more accurately, the cast is important in assignment, and the lack of a cast is important in the comparison.)